lucene/core/src/java/org/apache/lucene/search/DocValuesFieldExistsQuery.java

```diff
@@ -74,6 +74,16 @@ public Scorer scorer(LeafReaderContext context) throws IOException {
         return new ConstantScoreScorer(this, score(), scoreMode, iterator);
       }
 
+      @Override
+      public int count(LeafReaderContext context) throws IOException {
+        final LeafReader reader = context.reader();
+        FieldInfo fieldInfo = reader.getFieldInfos().fieldInfo(field);
+        if (fieldInfo == null || fieldInfo.getPointDimensionCount() == 0) {
+          return -1;
```

Comment:
Nit: In these cases we like to fall back to the default implementation, this way we don't have to know about what the contract is for the case when the count cannot be computed.
```suggestion
          return super.count(context);
```